backend/dataall/modules/worksheets/services/worksheet_services.py

```diff
@@ -0,0 +1,161 @@
+import logging
+
+from dataall import db
+from dataall.aws.handlers.sts import SessionHelper
+from dataall.core.permission_checker import has_tenant_permission, has_resource_permission
+from dataall.db import exceptions
+from dataall.db import models
+from dataall.db.api import ResourcePolicy
+from dataall.modules.worksheets.aws.athena_client import AthenaClient
+from dataall.modules.worksheets.db.models import Worksheet
+from dataall.modules.worksheets.db.repositories import WorksheetRepository
+from dataall.modules.worksheets.services.worksheet_permissions import MANAGE_WORKSHEETS, UPDATE_WORKSHEET, \
+    WORKSHEET_ALL, GET_WORKSHEET, DELETE_WORKSHEET, RUN_ATHENA_QUERY
+
+
+logger = logging.getLogger(__name__)
+
+
+class WorksheetService:
+    @staticmethod
+    def get_worksheet_by_uri(session, uri: str) -> Worksheet:
+        if not uri:
+            raise exceptions.RequiredParameter(param_name='worksheetUri')
+        worksheet = WorksheetRepository.find_worksheet_by_uri(session, uri)
+        if not worksheet:
+            raise exceptions.ObjectNotFound('Worksheet', uri)
+        return worksheet
+
+
+    @staticmethod
+    @has_tenant_permission(MANAGE_WORKSHEETS)
+    def create_worksheet(
+        session, username, groups, uri, data=None, check_perm=None
```

Comment:
If we start using `has_tenant_permission` (instead of `has_tenant_perm`) we can clean up the method parameters. It's not related to modularization, but we could change it if we are still re-working on this part

Reply:
good point, I'll do it, for the other functions as well